src/ml_tooling/baseclass.py

```diff
@@ -367,7 +291,7 @@ def test_estimators(
 
         return cls(best_estimator), ResultGroup(results)
 
-    def train_estimator(self, data: Dataset) -> "ModelData":
+    def train_estimator(self, data: Dataset) -> "Model":
```

Comment:
Why is the typehint in quotes?

Reply:
It's referring to itself, which is not defined yet, so quotes are valid for typehinting as well, in this case, but also if you can't import due to circular imports or something similar